src/parsers/power_system_table_data.jl

```diff
@@ -566,34 +567,55 @@ function services_csv_parser!(sys::System, data::PowerSystemTableData)
     bus_area_column = get_user_field(data, BUS::InputCategory, "area")
 
     # Shortcut for data that looks like "(val1,val2,val3)"
-    make_array(x) = split(strip(x, ['(', ')']), ",")
+    make_array(x) = isnothing(x) ? x : split(strip(x, ['(', ')']), ",")
+
+    function _add_device!(contributing_devices, device_categories, name)
+        for dev_category in device_categories
+            component_type = _get_component_type_from_category(dev_category)
+            components = get_components_by_name(component_type, sys, name)
+            if length(components) == 0
+                # There multiple categories, so we might not find a match in some.
+                continue
+            elseif length(components) == 1
+                component = components[1]
+            else
+                msg = "Found duplicate names type=$component_type name=$name"
+                throw(DataFormatError(msg))
+            end
+
+            push!(contributing_devices, component)
+        end
+    end
 
     for reserve in iterate_rows(data, RESERVE::InputCategory)
         device_categories = make_array(reserve.eligible_device_categories)
-        device_subcategories = make_array(reserve.eligible_device_subcategories)
+        device_subcategories = make_array(get(reserve, :eligible_device_subcategories, nothing))
+        devices = make_array(get(reserve, :contributing_devices, nothing))
         regions = make_array(reserve.eligible_regions)
+        requirement = get(reserve, :requirement, nothing)
         contributing_devices = Vector{Device}()
 
-        for gen in iterate_rows(data, GENERATOR::InputCategory)
-            bus_ids = data.bus[!, bus_id_column]
-            area = string(data.bus[bus_ids .== gen.bus_id, bus_area_column][1])
-            if gen.category in device_subcategories && area in regions
-                for dev_category in device_categories
-                    component_type = _get_component_type_from_category(dev_category)
-                    components = get_components_by_name(component_type, sys, gen.name)
-                    if length(components) == 0
-                        # There multiple categories, so we might not find a match in some.
-                        continue
-                    elseif length(components) == 1
-                        component = components[1]
-                    else
-                        msg = "Found duplicate names type=$component_type name=$name"
-                        throw(DataFormatError(msg))
+        if !isnothing(device_subcategories)
+            @info("Adding contributing generators for $(reserve.name) by category")
+            @warn("Adding contributing components by category only supports generators")
```

Comment:
It seems like this warning message could cause tons of noise.  Maybe add a maxlog=1?